src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -0,0 +1,46 @@
+package nextstep.courses.domain;
+
+import nextstep.users.domain.NextStepUser;
+import nextstep.users.domain.NextStepUsers;
+
+import java.time.LocalDateTime;
+
+public class Session {
+
+  private final Long id;
+  private final SessionPayment sessionPayment;
+  private SessionStatus sessionStatus;
+  private final MaxUserEnrollment maxUserEnrollment;
+  private final NextStepUsers nextStepUsers = new NextStepUsers();
+  private final SessionDate startDate;
+  private final SessionDate closedDate;
+  private final SessionCoverUrl sessionCoverUrl;
+
+  public Session(Long id, SessionPayment sessionPayment, SessionStatus sessionStatus, int maxUserEnrollment, LocalDateTime startDate, LocalDateTime closedDate, String sessionCoverUrl) {
+    this.startDate = new SessionDate(startDate);
+    this.startDate.validateClosedDate(closedDate);
+
+    this.id = id;
+    this.sessionPayment = sessionPayment;
+    this.sessionStatus = sessionStatus;
+    this.maxUserEnrollment = new MaxUserEnrollment(maxUserEnrollment);
+    this.closedDate = new SessionDate(closedDate);
+    this.sessionCoverUrl = new SessionCoverUrl(sessionCoverUrl);
+  }
+
+  public int currentEnrollmentCount() {
+    return nextStepUsers.currentEnrollment();
+  }
+
+  public void changeSessionStatus(SessionStatus sessionStatus) {
+    this.sessionStatus = sessionStatus;
+  }
+
+  public void processEnrollment(NextStepUser nextStepUser) {
+    sessionStatus.validateAcceptingStatus();
```

Comment:
`validateAcceptingStatus` 메서드명으로만 보면 AcceptingStatus 를 검증하는 메서드로 보여요! (상태가 잘못됐는지 확인하는 의도로 보여짐)
이런 접근방식도 좋지만, `validate`같은 메서드는 객체를 생성할 때 보통 사용합니당

```suggestion
    if(!sessionStatus.canEnroll()){
        throw new IllegalArgumentException();
    }
```

`Session`의 강의 상태는 수강 신청할수 없니? -> 예외
이런 방식으로 접근해보시는건 어떠신가요?

물론 재원님이 틀렸다는것은 절대 아닙니다! 😄 
다양한 접근방법이 있다는 리뷰를 드리고 싶었어용